pandas/tests/test_format.py

```diff
@@ -2952,6 +2952,16 @@ def test_to_csv_date_format(self):
         self.assertEqual(df_day.to_csv(), expected_default_day)
         self.assertEqual(df_day.to_csv(date_format='%Y-%m-%d'), expected_default_day)
 
+        # GH 7791
```

Comment:
right, they are NOT handled, that's why we need another issue :) ``date_format`` IS handled by the ``DatetimeIndex``, but was not being passed thru (same with ``nan_rep``)

But ``decimal/float_format`` need handling. These work for the actual data, just not the ``Index``.
(as I said not really that difficult; these apply ONLY to ``Float64Index``, so would need to have a ``_to_native_types`` for that index type that does basically what ``FloatBlock`` does).